packages/x-components/src/components/__tests__/preselected-filters.spec.ts

```diff
@@ -0,0 +1,99 @@
+import { mount } from '@vue/test-utils';
+import { createRawFilters } from '../../utils/filters';
+import { baseSnippetConfig } from '../../views/base-config';
+import PreselectedFilters from '../preselected-filters.vue';
+
+function renderPreselectedFilters({
+  filters,
+  snippetFilters
+}: RenderPreselectedFiltersOptions = {}): RenderPreselectedFiltersAPI {
+  const emit = jest.fn();
+  mount(PreselectedFilters, {
+    provide: {
+      snippetConfig: { ...baseSnippetConfig, filters: snippetFilters }
+    },
+    propsData: { filters },
+    mocks: {
+      $x: {
+        emit
+      }
+    }
+  });
+
+  return {
+    emit
+  };
+}
+
+describe('testing Preselected filters component', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('does not emit the event when neither filters nor snippet config filters are provided', () => {
+    const { emit } = renderPreselectedFilters();
+
+    expect(emit).not.toHaveBeenCalled();
+  });
+  it('emits the event when filters are provided in the snippet config', () => {
+    const snippetFilters = [
+      '{!tag=brand_facet}brand_facet:"Lego"',
+      '{!tag=age_facet}age_facet:"toddler"'
```

Comment:
Could you add how to pass this object in the [Snippet Configuration](https://github.com/empathyco/x/blob/main/packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md)

I understand this is an extra param, there is a row in the table for <extra parameters>, we could add in the description cell an example as the warehouse one.